### ISSUE TITLE
Index and slice a MultiScene

### ISSUE THREAD

**Issue by gerritholl** [pytroll/satpy#1426]
## Feature Request

**Is your feature request related to a problem? Please describe.**

I have a `MultiScene` with `ABI M1` data, i.e. data every minute.  I have a retrieval calculation that works on the difference between two following scenes.  To compare the effect of the time interval, I would like to slice/thin my `MultiScene` such that I can make my calculation every 2 minutes, every 5 minutes, etc.

**Describe the solution you'd like**

I would like to implement a `MultiScene.__getitem__` to allow for getting individual scenes (when passing an integer) or a new sliced multiscene (when passing a slice).

**Describe any changes to existing user workflow**

No backward compatibility issues, as there is currently no implementation for `MultiScene.__getitem__`.

**Additional context**

If `ms.scenes` (where `ms` is a `MultiScene` instance) is a sequence, then I can use `MultiScene(ms.scenes[::2])` to produce a `MultiScene` consisting of every second scene.  I'm not sure if this works in all cases or if some information may be lost in this case.

If we additionally implement `__len__`, we can mixin [`collections.abc.Sequence`](https://docs.python.org/3/library/collections.abc.html#collections.abc.Sequence) and get for free the methods `__contains__`, `__iter__`, `__reversed__`, `index`, and `count`.

Related issue on `Scene` objects: #1419

**Comment by djhoese** [comment]
So I'm a little torn/unsure on this. The original idea for why I structured the MultiScene the way I did and why it is so complicated, is that I wanted to avoid users completing the internal generator of the MultiScene by accident. The MultiScene tries to be smart with its methods by only creating generators based on generators whenever possible so Scenes are only created when necessary. The hope being that performance doesn't suffer and provide a bad user experience having to wait for 10s/100s/1000s of Scenes to be generated and a ton of files to be opened. I'm worried that some of these features will require iterating over the generator and make users question the performance of it all.

That said, I wonder if we could replace the Scene creation with dask Delayed objects. You'd still have to have a list (or can you have a numpy array of delayed objects?) with all the Scenes but created a Delayed object should be much lighter than creating the full Scene right away.

Now, all of **that** said, I'd love for the MultiScene and Scene to fit the generic interfaces defined in python (ex. Sequence, Mapping, etc).

**Comment by gerritholl** [comment]
Dask delayed objects sounds good.  I noticed this is surprisingly slow:

```python
from satpy import MultiScene, Scene
from glob import glob
m1 = MultiScene.from_files(glob("/data/gholl/cache/fogtools/abi/2020/04/12/01/C*/OR_ABI-L1b-RadM1-M6C*_G16_s202010301*.nc"), reader="abi_l1b")
for x in m1.scenes: pass
```

despite not having loaded anything, it's somewhat slow.  But that's a different issue from the one here.

**Comment by djhoese** [comment]
>  But that's a different issue from the one here.

Not really. This is exactly what I'm talking about. Iterating over the Scenes currently requires *creating* those Scenes. Up until that `for` loop the individual Scenes were waiting to be created in a generator. Creating them means parsing the filenames, passing the filenames to the file handler, each file handler has to open the file which for this reader requires xarray parsing a lot of information. This has to happen for 16 files for each Scene you are giving it. This doesn't require loading any actual data arrays so it shouldn't be dependent on the sector (full disk, conus, meso 1, meso 2) but is slow because of all the filename parsing and opening of files.
